cli/core/validator.go

```diff
@@ -2,17 +2,91 @@ package core
 
 import (
 	"context"
+	"fmt"
 	"math/big"
 	"strconv"
 
 	eigenpodproofs "github.com/Layr-Labs/eigenpod-proofs-generation"
 	"github.com/Layr-Labs/eigenpod-proofs-generation/cli/core/onchain"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/fatih/color"
 )
 
-func GenerateValidatorProof(ctx context.Context, eigenpodAddress string, eth *ethclient.Client, chainId *big.Int, beaconClient BeaconClient) (*eigenpodproofs.VerifyValidatorFieldsCallParams, []uint64) {
+func SubmitValidatorProof(ctx context.Context, owner, eigenpodAddress string, chainId *big.Int, eth *ethclient.Client, batchSize uint64, proofs *eigenpodproofs.VerifyValidatorFieldsCallParams, noPrompt bool) ([]*types.Transaction, error) {
+
+	ownerAccount, err := PrepareAccount(&owner, chainId)
+	if err != nil {
+		return nil, err
+	}
+	PanicOnError("failed to parse private key", err)
+
+	eigenPod, err := onchain.NewEigenPod(common.HexToAddress(eigenpodAddress), eth)
+	if err != nil {
+		return nil, err
+	}
+
+	indices := Uint64ArrayToBigIntArray(proofs.ValidatorIndices)
+	validatorIndicesChunks := chunk(indices, batchSize)
+	validatorProofsChunks := chunk(proofs.ValidatorFieldsProofs, batchSize)
+	validatorFieldsChunks := chunk(proofs.ValidatorFields, batchSize)
+	if !noPrompt {
+		PanicIfNoConsent(fmt.Sprintf("This will call EigenPod.VerifyWithdrawalCredentials() %d times, to link your validator to your eigenpod.", len(validatorIndicesChunks)))
+	}
+
+	color.Green("calling EigenPod.VerifyWithdrawalCredentials() (using %d txn(s), max(%d) proofs per txn)", len(indices), batchSize)
+
+	latestBlock, err := eth.BlockByNumber(ctx, nil)
```

Comment:
wait what? this needs to be calling the EigenPod contract

call https://github.com/Layr-Labs/eigenlayer-contracts/blob/6321f161162cb2dced8543b0f2e86d7bc23abbdc/src/contracts/pods/EigenPod.sol#L693

with the timestamp of the beacon state the proofs are made against

Reply:
omg lol

Reply:
hold on

Reply:
this is actually wrong because of a race condition - proofs are being generated in `GenerateValidatorProof` -- that probably needs to return a timestamp rather than using `lastestBlock` twice

Reply:
But `GenerateValdiatorProof` is using `latestBlock` correctly, i think.

Reply:
isn't `oracleBeaconTimestamp` unused in these proofs? i thought you had me put up a PR to remove it from the proofgen lol

Reply:
discussed offline, will be piping `latestBlock.Time` through from `GenerateValidatorProof`!